TPP.Core/Modes/Matchmode.cs

```diff
@@ -132,5 +143,34 @@ public void Dispose()
         {
             _modeBase.Dispose();
         }
+
+        public async void ResetBalances()
+        {
+            _logger.LogInformation("Resetting Balances");
+            long minimumPokeyen = _matchmodeConfig.MinimumPokeyen;
+            long subscriberMinimumPokeyen = _matchmodeConfig.SubscriberMinimumPokeyen;
+
+            List<User> poorUsers = await _userRepo.ListAllUnderPokeyen(Math.Max(minimumPokeyen, subscriberMinimumPokeyen));
+            foreach (User u in poorUsers)
+            {
+                long pokeyen = await _pokeyenBank.GetAvailableMoney(u);
+                if(u.IsSubscribed && pokeyen < subscriberMinimumPokeyen)
+                {
+                    long amountToGive = subscriberMinimumPokeyen - pokeyen;
+                    await _pokeyenBank.PerformTransaction(new Transaction<User>(u, amountToGive, TPP.Persistence.Repos.TransactionType.Welfare));
+                    // TODO whisper users informing them they have been given money
+                    _logger.LogInformation(String.Format("Subscriber {0} had their balance reset to P{1} (+P{2})", u.SimpleName, subscriberMinimumPokeyen, amountToGive));
+
+                }
+                else if (!u.IsSubscribed && u.Pokeyen < minimumPokeyen)
+                {
+                    long amountToGive = minimumPokeyen - pokeyen;
+                    await _pokeyenBank.PerformTransaction(new Transaction<User>(u, amountToGive, TPP.Persistence.Repos.TransactionType.Welfare));
```

Comment:
- `TPP.Persistence.Repos.TransactionType.Welfare` looks over-qualified
- the opportunity arises to use `PerformTransactions` (plural) to bundle all balance reset operations